homeassistant/components/reolink/host.py

```diff
@@ -386,10 +430,40 @@ def unregister_webhook(self):
         webhook.async_unregister(self._hass, self.webhook_id)
         self.webhook_id = None
 
+    async def _async_long_polling(self, *_) -> None:
+        """Use ONVIF long polling to immediately receive events."""
+        # This task will be cancelled once _async_stop_long_polling is called
+        while True:
+            if self._webhook_reachable:
+                self._long_poll_task = None
+                await self._async_stop_long_polling()
+                return
+
+            try:
+                channels = await self._api.pull_point_request()
+            except Exception as ex:  # pylint: disable=broad-except
+                if not self._long_poll_error:
+                    _LOGGER.exception("Error while requesting ONVIF pull point: %s", ex)
+                    await self._api.unsubscribe(sub_type=SubType.long_poll)
+                self._long_poll_error = True
+                await asyncio.sleep(30)
+                continue
+
+            self._long_poll_error = False
+
+            if not self._long_poll_received and channels != []:
+                self._long_poll_received = True
+                ir.async_delete_issue(self._hass, DOMAIN, "webhook_url")
+
+            self._signal_write_ha_state(channels)
+
+            # Cooldown to prevent CPU over usage on camera freezes
+            await asyncio.sleep(0.75)
+
     async def _async_poll_all_motion(self, *_) -> None:
         """Poll motion and AI states until the first ONVIF push is received."""
-        if self._webhook_reachable.is_set():
-            # ONVIF push is working, stop polling
+        if self._webhook_reachable or self._long_poll_received:
+            # ONVIF push or long polling is working, stop fast polling
```

Comment:
Are there cameras that don't support pull point?  Ideally we can get rid of the complexity of having 3 methods to get the events